### PR TITLE
Moved the entire static site to Hugo

### PR DESCRIPTION
**Pull request by VoyTechnology**
Main benefit is that we don't have to do weird checking out of gh-pages
and its easier to modify certain pages.

Visually, there is very little difference. Sure my CSS could be a bit better but that is easy to fix.

Instructions on enabling the CI to build:
1. Setup project on CircleCI: https://circleci.com/add-projects/gh/iandioch/cpssd.net
2. Authorize CircleCI to use personal git key to push to the repo (`gh-pages`) branch: https://circleci.com/gh/andioch/cpssd.net/edit#checkout

Feel free to PM me or add comments below.

### CODE REVIEW

**Comment by VoyTechnology** [comment]
This change requires CircleCI to be set up in order to deploy the pages to `gh-pages`

**Comment by byxor** [comment]
I don't know what any of this means, but it seems like a big change. Weren't we already using Hugo to generate the website?

**Comment by VoyTechnology** [comment]
Currently you need to checkout `gh-pages`, and the site is being generated by Jekyll. If you want to try and develop anything locally you need to download ruby and all its dependencies (https://jekyllrb.com/docs/installation/), Hugo comes as a standard 5MB statically compiled executable (https://gohugo.io/getting-started/installing/) and you can test locally with `hugo server`

**Comment by byxor** [comment]
I don't see any issues without checking out `gh-pages`. Although not requiring a bunch of external dependencies seems nice.

**Comment by iandioch** [comment]
This is a lot of complexity being added, but I'm not sure what the benefit is. Everyone is familiar with gh-pages and how it works. How often do you actually need to develop locally?

I won't get the chance before next week to try running it and seeing how it works, will report back when I do.

**Comment by VoyTechnology** [comment]
The only additional complexity is CircleCI, and only for @iandioch, once, and even that can be omitted and the scripts cane be ran by hand. 

A more optimal approach overall would be to have a separate org with multiple maintainers, and moving the repo there. But that is its own issue.

**Comment by byxor** [comment]
If this gets merged, is the entire site going to break? I want to submit a PR but I'm holding off until this PR is dealt with.

**Comment by VoyTechnology** [comment]
@sentriz made a PR to my PR and he was very happy with the simplicity 

(right @sentriz? Plz say it was easier to add data to Hugo rewrite than current site, plz, plz)

**Comment by byxor** [comment]
@VoyTechnology It's late, you should go to bed. :wink:

**Comment by VoyTechnology** [comment]
@byxor 99% sure it will. But I'll be happy to update this PR if you make and merge your PR to the current site.

File: LICENSE.md [hunk 1]
```diff
@@ -1,6 +1,4 @@
-## Released under MIT Licence
-
-The Emerald theme for Jekyll is Copyright (c) 2015 Jacopo Rabolini
+Copyright (c) 2018 CPSSD Students
```
Comment (iandioch) [inline LICENSE.md]: Disagree with this file rename. Regardless, there's no reason why this change should be part of this PR - put it in a different one.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+---
+
+## Contributing
+
+Feel free to submit pull requests, make issues, and edit anything and everything.
+
+Anyone can add a post to the blog. We'll figure out some way of organising that as it goes. Just copy the format of an existing post and submit a pull request.
+
+Anybody can add a post to the blog. Just create a new file in `content/blog/`,
```
Comment (iandioch) [inline README.md]: This paragraph and the previous paragraph need to be merged.

File: config.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+baseURL: https://cpssd.net
+disablePathToLower: true
```
Comment (iandioch) [inline config.yaml]: y tho

File: config.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+languageCode: en-us
```
Comment (iandioch) [inline config.yaml]: en-ie

File: content/about.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+we mention its now gone too?
+
+This site exists to give friendlier introduction to CPSSD, from the viewpoint
+of the students, remembering how good of a course it was. It was not and is not
+in any way official or endorsed by DCU.
+
+[1]: https://dcu.ie
+[2]: /projects
+[3]: /blog/rip
+[4]: https://web.archive.org/web/20170720163904/http://www.computing.dcu.ie/undergraduate/pssd/computational-problem-solving-software-development-cpssd
```
Comment (iandioch) [inline content/about.md]: +1 to web.archive.org

File: content/blog/2017-hashcode.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+---
+date: "2017-02-26T00:00:00Z"
+title: CPSSD Tackles Hashcode 2017
+---
+
+Google's [Hashcode][1] qualification round took
+place on the 23rd of February, and lots of CPSSD first and second years crowded
+in the DCU Hashcode hub to compete.
+
+<!--more-->
```
Comment (iandioch) [inline content/blog/2017-hashcode.md]: Does this comment do something?

File: static/css/main.scss [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+# Front matter comment to ensure Jekyll properly reads file.
```
Comment (iandioch) [inline static/css/main.scss]: Jekyll?

File: themes/cpssd/static/css/main.css [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  font-size: 3rem;
+}
+
+header::after {
+  display: block;
+  content: '';
+  width: 15%;
+  height: 1px;
+  margin: 1rem auto;
+  background-color: lightgrey;
```
Comment (iandioch) [inline themes/cpssd/static/css/main.css]: does `lightgrey` really work instead of `lightgray`?

File: themes/cpssd/static/css/main.css [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+.title {
+  font-size: 2rem;
+  margin: 0;
+  margin-bottom: 0;
+}
+
+.date {
+  color: grey;
```
Comment (iandioch) [inline themes/cpssd/static/css/main.css]: see above re: `grey`/`gray`.

File: themes/cpssd/static/css/main.css (hunk 8 above)
Comment (byxor) [reply themes/cpssd/static/css/main.css]: https://www.w3schools.com/colors/color_tryit.asp?color=LightGrey

Comment (iandioch) [reply themes/cpssd/static/css/main.css]: [CSS took colour names from X11, including duplicates](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value). So yeah, it works, TIL.

File: content/blog/2017-hashcode.md (hunk 6 above)
Comment (VoyTechnology) [reply content/blog/2017-hashcode.md]: Yes. It allows you to define the end of a summary. So if I have a long article, and I  place `<!--more-->` after the first paragraph, the article list would show only the first paragraph, and you will have to click on the article to view the whole thing.

File: config.yaml (hunk 3 above)
Comment (VoyTechnology) [reply config.yaml]: This was some time ago that I did it. I believe I imported the site from Jekyll and it was the default setting. I believe this basically means that if a file is called `SomeBlogPost.md`, this would generate `SomeBlogPost.html` from it instead of `someblogpost.html`. I can remove that line if you want.

**Comment by VoyTechnology** [comment]
PTAL @iandioch

File: config.yaml (hunk 3 above)
Comment (iandioch) [reply config.yaml]: It sounds like nbd

File: README.md [hunk 10]
```diff
@@ -22,11 +22,11 @@ hugo server
 
 Feel free to submit pull requests, make issues, and edit anything and everything.
 
-Anyone can add a post to the blog. We'll figure out some way of organising that as it goes. Just copy the format of an existing post and submit a pull request.
-
-Anybody can add a post to the blog. Just create a new file in `content/blog/`,
-probably prefix it with a year if its a recurring event or something. You can
-just copy the definition at the top from another file.
+Anyone can add a post to the blog. We'll figure out some way of organising that
+as it goes. Just copy the format of an existing post and submit a pull
```
Comment (iandioch) [inline README.md]: When I meant they should be merged, I didn't just mean stuck together 😛 The two paragraphs are near duplicates ("Anyone can add a post to the blog", then later, "Anybody can add a post to the blog").

Just de-dupe, and go with something like:

"Anyone can add a post to the blog. Just create a new file in `content/blog`, probably prefix it with a year if it's a recurring event or something. You can just copy the definition at the top from another file. Then submit a pull request."

**Comment by VoyTechnology** [comment]
PTAL @iandioch, hopefully for the last time :)

**Comment by byxor** [comment]
🍿

**Comment by sentriz** [comment]
how2fix

**Comment by VoyTechnology** [comment]
@sentriz your issue is now fixed (I was using old version of hugo), @iandioch Everything is ready for final approval and merge.

**Comment by iandioch** [comment]
TY - will take a look after work this evening xo

**Comment by sentriz** [comment]
<p align="center">

</p>

**Comment by VoyTechnology** [comment]
Please report any issues on #34
